stl_text/ops/tokenizers/whitespace_tokenizer.py

```diff
@@ -1,44 +1,57 @@
+import torch
 import torch.nn as nn
 from typing import Dict, List, Optional
+from collections import OrderedDict
 from iopath.common.file_io import PathManager
-
+from torchtext.experimental.vocab import (
+    load_vocab_from_file,
+    Vocab,
+)
+from torchtext._torchtext import Vocab as VocabPybind 
 
 class WhitespaceTokenizer(nn.Module):
-    def __init__(self, vocab_path: Optional[str] = None, trainable=False, speed: int = 0):
+    """
+    from stl_text.ops.tokenizers import WhitespaceTokenizer
+    whitespacetokenizer = WhitespaceTokenizer(vocab_path=vocab_path)
+    jit_whitespacetokenizer = torch.jit.script(whitespacetokenizer)
+    print(jit_whitespacetokenizer.code)
+
+    >>>>>>>>>>>>>>>>>>>>>>>>>>OUTPUT of JIT code<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+    def forward(self,
+        text: str) -> List[int]:
+    _0 = (self).numberize(torch.split(text, None, -1), )
+    return _0
+    >>>>>>>>>>>>>>>>>>>>>>>>>>OUTPUT of JIT code<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
+
+    """
+
+    def __init__(self, vocab_path: Optional[str] = None, trainable: bool = False):
         super(WhitespaceTokenizer, self).__init__()
         self.trainable = trainable
-        self.speed = speed  # mock a real tokenizer: slowing down tokenization speed
-
-        self.unknown = "unknown"
-        self.vocab: Dict[str, int] = {self.unknown: 0}
+        self.unk_token = '<unk>'
 
-        # load vocab
-        path_manager = PathManager()
         if vocab_path:
-            with path_manager.open(vocab_path, "r") as f:
-                for line in f.readlines():
-                    token = line.split()[0]
-                    self.vocab[token] = len(self.vocab)
+            path_manager = PathManager()
+            with path_manager.open(vocab_path, "r",encoding='utf-8') as f:
+                self.vocab = load_vocab_from_file(f,unk_token=self.unk_token)
+        else:
+            self.vocab = Vocab(VocabPybind([self.unk_token],self.unk_token))
 
-    def forward(self, text: str) -> List[int]:
-        return self.numberize(self.tokenize(text))
+        self.vocab = self.vocab.to_ivalue() #Remove to_ivalue() PR: https://github.com/pytorch/text/pull/1080
```

Comment:
On the eager mode, we suggest to use the pybind vocab (aka, not to call `to_ivalue`) for better performance. `to_ivalue` is called to switch to torchbind vocab before calling JIT.